lib/msal-common/src/request/CommonClientCredentialRequest.ts

```diff
@@ -19,4 +20,5 @@ export type CommonClientCredentialRequest = BaseAuthRequest & {
     skipCache?: boolean;
     azureRegion?: AzureRegion;
     clientAssertion?: ClientAssertion;
+    tokenQueryParameters?: StringDict;
```

Comment:
I think these should go to all flows. At least all confidential client flows, but I think all public client flows would benefit as well. This allows app developers to select an AAD test slice by adding a special param as query param to the token call (which is a POST request, but it still works). 

@tnorling @peterzenz @sameerag - are you ok with adding this for all flows, i.e in `BaseAuthRequest` ?

Reply:
This is already implemented for PublicClient flows but I have no objections to adding it where it's missing or moving it to be more centrally available

Reply:
@tnorling - no objection here, but please help me understand the inheritance diagram. I see that [DeviceCode](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/9384b584a3383732d9070bd7ed345589f8969c3f/lib/msal-common/src/request/CommonDeviceCodeRequest.ts), [ROPC](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/9384b584a3383732d9070bd7ed345589f8969c3f/lib/msal-common/src/request/CommonUsernamePasswordRequest.ts) and [silent](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/9384b584a3383732d9070bd7ed345589f8969c3f/lib/msal-common/src/request/CommonSilentFlowRequest.ts) all extend `BaseAuthRequest` and not `CommonAuthorizationUrlRequest`

Reply:
You're right, I should have clarified that all of our flows that use _at least_ the `/authorize` endpoint (which is indeed most, but not all PCA APIs) use this type. This will indeed need to be extended at least to Silent. DeviceCode may need its own parameter given it doesn't use `/token`. I'm less certain about the value of adding this for ROPC, but no objections if you'd like to do it anyway. 

One thing I'm remembering now as we're talking about this is that we didn't add `tokenQueryParameters` (or `extraQueryParameters`) to `BaseAuthRequest` because not all request types ultimately make a `/token` request. To keep the request type surface clean I'd ask that you use one of two approaches:

1. Move the params to `BaseAuthRequest` but then omit them from downstream requests where they are not relevant
or
2. Only include the params on the types which will actually use them

We initially went with option 2 but it appears we either didn't do it completely or some of these types didn't exist yet at that time. If you want to switch to approach 1 I'm ok with that.